MathPortal/MathPortal/Source/Tools/Appearence.swift

```diff
@@ -9,7 +9,26 @@
 import UIKit
 
 class Appearence {
+    
+    static func getGradientLayerFor(_ myView: UIView?, colors: [CGColor]) -> CAGradientLayer? {
```

Comment:
Not that it is wrong but I discourage you to use shape layers as a whole like this. The problem is that they do not use constraints and are very hard to maintain. It is better to subclass a view and put a gradient on it. I have usually been using this:

```
@IBDesignable class VerticalGradientView: UIView {
    
    @IBInspectable var topColor: UIColor? {
        didSet {
            setNeedsDisplay()
        }
    }
    @IBInspectable var bottomColor: UIColor? {
        didSet {
            setNeedsDisplay()
        }
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        setNeedsDisplay()
    }
    
    override func draw(_ rect: CGRect) {
        super.draw(rect)
        
        guard let context = UIGraphicsGetCurrentContext(), let topColor = topColor, let bottomColor = bottomColor else {
            return
        }
        
        let locations: [CGFloat] = [0.0, 1.0]
        let colors = [topColor.cgColor, bottomColor.cgColor]
        if let gradient = CGGradient(colorsSpace: CGColorSpaceCreateDeviceRGB(), colors: colors as CFArray, locations: locations) {
            context.drawLinearGradient(gradient, start: CGPoint.zero, end: CGPoint(x: 0.0, y: frame.size.height), options: CGGradientDrawingOptions(rawValue: 0))
        }
    }
    
}

```

I think in iOS channel I posted a toolset some time ago that should include GradientViews files which has a few of these...